src/models/Component.js

```diff
@@ -204,6 +282,43 @@ class Component extends FileInformation {
     return null;
   }
 
+  /**
+   * Set attributes corresponding to the given attribute field in the provided array.
+   * Search in sub-attributes of "Object" attributes also.
+   * @param {ComponentAttribute[]} result - The array to store the found attributes.
+   * @param {ComponentAttribute[]} attributes - Attributes list.
+   * @param {string} field - Attribute field to check.
+   * @param {...string} values - Field values to compare.
+   * @private
+   */
+  __setAttributesByField(result, attributes, field, ...values) {
+    attributes.forEach((attribute) => {
+      if (attribute?.type === 'Object') {
+        this.__setAttributesByField(result, attribute.value, field, ...values);
+      }
+
+      if (this.__compareAttributeField(attribute, field, ...values)) {
+        result.push(attribute);
+      }
+    });
+  }
+
+  /**
+   * Compare attribute field to given value and indicate if its contains or not.
```

Comment:
..indicate if the value contains that attribute or not.

Reply:
done